snuba/request/schema.py

```diff
@@ -78,8 +80,14 @@ def build_with_extensions(
         cls,
         extensions: Mapping[str, QueryExtension],
         settings_class: Type[RequestSettings],
+        language: str,
     ) -> RequestSchema:
-        generic_schema = GENERIC_QUERY_SCHEMA
+
```

Comment:
Up to you, I would review it anyway, but if I were you, I would split this PR into the changes to the parser to support the different syntax, the support for match, and exposing the parser via the snuba endpoint. Being these fairly independent features, it would make the review process quicker (fewer back and forth and more frequent) and it would make it easier to resolve conflicts.